foodsaving/groups/tests/test_tasks.py

```diff
@@ -0,0 +1,47 @@
+from django.db import DataError
+from django.db import IntegrityError
+from django.test import TestCase
+from django.core import mail
+from django.utils import timezone
+from rest_framework.test import APITestCase
+
+from foodsaving.groups.factories import GroupFactory
+from foodsaving.groups.models import Group, GroupMembership
+from foodsaving.groups.tasks import process_inactive_users
+from foodsaving.users.factories import UserFactory
+from config import settings
+
+from datetime import timedelta
+
+
+# This probably shouldn't be APITestCase???
+class TestProcessInactiveUsers (APITestCase):
+    def setUp(self):        
+        self.admin = UserFactory()  # has membership management rights
+        self.member1 = UserFactory()
+        self.member2 = UserFactory()
+        self.group = GroupFactory(members=[self.admin, self.member1, self.member2])
+
+        now = timezone.now()
+
+        inactive_email_date = now - timedelta(days=settings.NUMBER_OF_DAYS_UNTIL_INACTIVE_IN_GROUP)
+        self.membership1 = GroupMembership.objects.get(group=self.group, user=self.member1)
+        self.membership1.lastseen_at = inactive_email_date
+        self.membership1.save()
+
+        remove_email_date = now - timedelta(days=settings.NUMBER_OF_DAYS_UNTIL_REMOVED_FROM_GROUP+1)
+        self.membership2 = GroupMembership.objects.get(group=self.group, user=self.member2)
+        self.membership2.lastseen_at = remove_email_date
+        self.membership2.active = False
+        self.membership2.save()
+
+        mail.outbox = []
+
+    def test_process_user_inactive(self):
+        process_inactive_users()
+        self.assertEqual(len(mail.outbox), 2)
```

Comment:
You should probably check if the user got the right e-mail. You can access the subject of an email with `mail.outbox[0].subject` and compare it to a string. For example:

```
member1_email = next(email for email in self.outbox if email.to[0] == self.member1.email)
self.assertIn('Please come back to', member1.subject)
```

I should probably write a helper to assist finding the email to a user...

Reply:
Checking the content of emails is reasonable. The contents of the emails should probably be agreed first though.